vars/runInsidePod.groovy

```diff
@@ -0,0 +1,80 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.ConfigurationLoader
+import com.sap.piper.SysEnv
+import java.util.UUID
+
+def call(Map parameters = [:], body) {
+    def uniqueId = UUID.randomUUID().toString()
+
+    ConfigurationHelper configurationHelper = new ConfigurationHelper(parameters)
+    def script = configurationHelper.getConfigProperty('script')
+    def containersMap = configurationHelper.getConfigProperty('containersMap',[:])
+    def dockerEnvVars = configurationHelper.getConfigProperty('dockerEnvVars',[:])
+    def dockerWorkspace = configurationHelper.getConfigProperty('dockerWorkspace','')
+
+    handleStepErrors(stepName: 'runInsidePod', stepParameters: [:]) {
+        def options = [name      : 'dynamic-agent-' + uniqueId,
+                       label     : uniqueId,
+                       containers: getContainerList(script,containersMap,dockerEnvVars, dockerWorkspace)]
+        podTemplate(options) {
+            node(uniqueId) {
+                body()
+            }
+        }
+    }
+}
+
+private getContainerList(script, containersMap,dockerEnvVars, dockerWorkspace ) {
+    def envVars
+    def jnlpAgent = ConfigurationLoader.generalConfiguration(script).jnlpAgent ?: 's4sdk/jenkins-agent-k8s:latest'
```

Comment:
Do we need a particular JNLP image from s4sdk or could it be the standard image from the plugin, too (would avoid maintenance costs for the s4sdk image)?

Reply:
We have a dependency on use id inside a docker container (1000). If we have one with user ID 1000. We can use that image.

Reply:
Updated the PR. Thanks for pointing it out.